docs/how-to-build-with-ddn/with-mongodb.mdx

```diff
@@ -0,0 +1,397 @@
+schema.
+
+<Prereqs />
+
+:::info Additionally, you'll need mongosh
+
+This tutorial uses mongosh — the Mongo shell — to interact with the local MongoDB instance. You can download it
+[here](https://www.mongodb.com/try/download/shell).
+
+:::
+
+## Tutorial
+
+### Step 1. Authenticate your CLI
+
+```sh title="Before you can create a new Hasura DDN project, you need to authenticate your CLI:"
+ddn auth login
+```
+
+This will launch a browser window prompting you to log in or sign up for Hasura DDN. After you log in, the CLI will
+acknowledge your login, giving you access to Hasura Cloud resources.
+
+### Step 2. Scaffold out a new local project
+
+```sh title="Next, create a new local project:"
+ddn supergraph init my-project && cd my-project
+```
+
+Once you move into this directory, you'll see your project scaffolded out for you. You can view the structure by either
+running `ls` in your terminal, or by opening the directory in your preferred editor.
+
+### Step 3. Initialize your MongoDB connector
+
+```sh title="In your project directory, run:"
+ddn connector init my_mongo -i
+```
+
+From the dropdown, start typing `mongo` and hit enter to accept the default port. Then, provide the following connection
+string:
+
+```plaintext
+mongodb://local.hasura.dev:27017/my_database
+```
+
+### Step 4. Start the local MongoDB container
+
+```sh title="Begin by creating a compose file for the Mongo service:"
+touch app/connector/my_mongo/compose.mongo.yaml
+```
+
+```yaml title="Then, open the file and add the following:"
+services:
+  mongodb:
+    image: mongo:latest
+    container_name: mongodb
+    ports:
+      - "27017:27017"
+```
+
+```sh title="Run the container:"
+docker compose -f app/connector/my_mongo/compose.mongo.yaml up -d
+```
+
+```sh title="Use the mongosh shell to seed the database:"
+docker exec -it mongodb mongosh my_database --eval "
+db.users.insertMany([
+  { user_id: 1, name: 'Alice', age: 25 },
+  { user_id: 2, name: 'Bob', age: 30 },
+  { user_id: 3, name: 'Charlie', age: 35 }
+]);
+"
+```
+
+The shell will return information about the newly-inserted `users` records.
+
+### Step 6. Introspect your MongoDB database
+
+```sh title="Next, use the CLI to introspect your MongoDB database:"
+ddn connector introspect my_mongo
+```
+
+After running this, you should see a representation of your collection's schema in
+`app/connector/my_mongo/schema/users.json`; you can view this using `cat` or open the file in your editor.
+
+For each collection in your database, the MongoDB connector will generate a separate JSON file representing it.
+
+### Step 7. Add your model
+
+```sh title="Now, track the collection from your MongoDB database as a model:"
+ddn models add my_mongo users
+```
+
+Open the `app/metadata` directory and you'll find a newly-generated file: `Users.hml`. The Hasura engine will use this
+Hasura Metadata Language file to bring the `users` table from MongoDB into your API as a
+[model](/supergraph-modeling/models.mdx).
```

Comment:
> The Hasura engine will use this Hasura Metadata Language file to bring the `users` table from MongoDB into your API as a [model](/supergraph-modeling/models.mdx).

This doesn't feel accurate enough. The *DDN CLI* takes hml metadata files with objects and creates a big JSON blobs (builds) for the Hasura engine to run as an API. 

If we're wanting to say *what happened* then the DDN CLI took the introspection of the collection and created an HML model representation for it in the supergraph. 

If we're wanting to say *what's about to happen* then the DDN CLI will take the hml model representation of the mongo collection and create a JSON blob build for the Hasura engine to run. (which also contains models etc) 

If it's what's about to happen my suggestion would be something like:

```suggestion
Open the `app/metadata` directory and you'll find a newly-generated file: `Users.hml`. The DDN CLI will now use this Hasura Metadata Language file to represent the `users` collection from MongoDB in your API as a
[model](/supergraph-modeling/models.mdx).
```

But could easily be what happened too, not what's about to happen. 

Also table == collection for mongo

Noting that we have the same kind of issue with the postgres version too.

Reply:
✅ Fixed in PG, too.